src/components/jacobian_component/jacobian.sh

```diff
@@ -70,74 +80,90 @@ setup_jacobian() {
 	rm -rf gcclassic
 	ln -s ${RunTemplate}/gcclassic .
 
-    # Link to restart file
-    RestartFileFromSpinup=${RunDirs}/spinup_run/Restarts/GEOSChem.Restart.${SpinupEnd}_0000z.nc4
-    if test -f "$RestartFileFromSpinup" || "$DoSpinup"; then
-        ln -s $RestartFileFromSpinup Restarts/GEOSChem.Restart.${StartDate}_0000z.nc4
+	# Link to restart file
+	RestartFileFromSpinup=${RunDirs}/spinup_run/Restarts/GEOSChem.Restart.${SpinupEnd}_0000z.nc4
+	if test -f "$RestartFileFromSpinup" || "$DoSpinup"; then
+            ln -s $RestartFileFromSpinup Restarts/GEOSChem.Restart.${StartDate}_0000z.nc4
 	else
 	    RestartFile=${RestartFilePrefix}${StartDate}_0000z.nc4
 	    ln -s $RestartFile Restarts/GEOSChem.Restart.${StartDate}_0000z.nc4
 	    if "$UseBCsForRestart"; then
 		    sed -i -e "s|SpeciesRst|SpeciesBC|g" HEMCO_Config.rc
-        fi
+            fi
 	fi
    
 	# Update settings in geoschem_config.yml except for the base run
-    if [ $x -ne 0 ]; then
+	if [ $x -ne 0 ]; then
 	    sed -i -e "s|emission_perturbation_factor: 1.0|emission_perturbation_factor: ${PerturbValue}|g" \
 	           -e "s|state_vector_element_number: 0|state_vector_element_number: ${xUSE}|g" geoschem_config.yml
-    fi
+	fi
 
-    # BC optimization setup
-    if "$OptimizeBCs"; then
-        bcThreshold=$(($nElements - 4))
-        # The last four state vector elements are reserved for BC optimization of NSEW
-        # domain edges. If the current state vector element is one of these, then
-        # turn on BC optimization for the corresponding edge and revert emission perturbation
-        if [ $x -gt $bcThreshold ]; then
-            PerturbBCValues=$(generate_BC_perturb_values $bcThreshold $x $PerturbValueBCs)
-            sed -i -e "s|CH4_boundary_condition_ppb_increase_NSEW:.*|CH4_boundary_condition_ppb_increase_NSEW: ${PerturbBCValues}|g" \
-                -e "s|perturb_CH4_boundary_conditions: false|perturb_CH4_boundary_conditions: true|g" \
-                -e "s|emission_perturbation_factor: ${PerturbValue}|emission_perturbation_factor: 1.0|g" \
-                -e "s|state_vector_element_number: ${xUSE}|state_vector_element_number: 0|g" geoschem_config.yml
-        fi
-    fi 
+	# BC optimization setup
+	if "$OptimizeBCs"; then
+	    if "$OptimizeOH"; then
```

Comment:
Nope. First there is a check for OptimizeBCs and if that's on it will also check if optimizeOH is on, and if so, it will account for OH as the last element. After that if statement, there's another one for OptimizeBCs.

```
	# BC optimization setup
	if "$OptimizeBCs"; then
	    if "$OptimizeOH"; then
		bcThreshold=$(($nElements - 5))
            else
		bcThreshold=$(($nElements - 4))
	    fi
            # The last four state vector elements are reserved for BC optimization of NSEW
            # domain edges. If the current state vector element is one of these, then
            # turn on BC optimization for the corresponding edge and revert emission perturbation
            if [ $x -gt $bcThreshold ]; then
		PerturbBCValues=$(generate_BC_perturb_values $bcThreshold $x $PerturbValueBCs)
		sed -i -e "s|CH4_boundary_condition_ppb_increase_NSEW:.*|CH4_boundary_condition_ppb_increase_NSEW: ${PerturbBCValues}|g" \
                       -e "s|perturb_CH4_boundary_conditions: false|perturb_CH4_boundary_conditions: true|g" \
                       -e "s|emission_perturbation_factor: ${PerturbValue}|emission_perturbation_factor: 1.0|g" \
                       -e "s|state_vector_element_number: ${xUSE}|state_vector_element_number: 0|g" geoschem_config.yml
            fi
	fi

	if "$OptimizeOH"; then
	    # The last state vector element is reserved for OH optimization.
            # If this is the current state vector element, then modify the OH
            # perturb value in HEMCO_Config.rc and revert emission perturbation.
	    OHthreshold=$(($nElements - 1))
            if [ $x -gt $OHthreshold ]; then
		sed -i -e "s|emission_perturbation: ${PerturbValue}|emission_perturbation: 1.0|g" \
                       -e "s|state_vector_element_number: ${xUSE}|state_vector_element_number: 0|g" geoschem_config.yml
		sed -i -e "s| OH_pert_factor  1.0| OH_pert_factor  ${PerturbValueOH}|g" HEMCO_Config.rc
            fi
	fi
```

The OH is global here. I was going to submit the hemispheric update separately but I can add that on top of these updates today just to get it in.